### ISSUE TITLE
Items missing from iCloud Drive locations

### ISSUE THREAD

**Issue by jbmorley** [inseven/opolua#39]
Only locally downloaded files appear in iCloud Drive locations. We should support showing remote files and downloading them when required.

**Comment by jbmorley** [comment]
We should definitely address this for v1 as it's likely that users will use iCloud, and will expect it to 'Just Work'.

**Comment by jbmorley** [comment]
It looks like this is going to be a bit of a big one, since we need to use an entirely separate API to index iCloud locations as files that haven't been requested locally don't turn up in the `FileManager` operations.

- https://stackoverflow.com/questions/21094534/how-do-i-verify-a-files-existence-in-icloud
- https://developer.apple.com/library/archive/documentation/Carbon/Conceptual/SpotlightQuery/Concepts/QueryingMetadata.html

**Comment by jbmorley** [comment]
It looks like `NSMetadataQuery` doesn't actually work for this, but it seems we can find the `.XXX.icloud` placeholder files using `FileManager.default.enumerator` and then request downloads for those. It's pretty ugly, but it should work.

For example,

```swift
let enumerator = FileManager.default.enumerator(at: self, includingPropertiesForKeys: nil)
while let item = enumerator?.nextObject() as? URL {
    if item.pathExtension == "icloud" {
        try? FileManager.default.startDownloadingUbiquitousItem(at: item)
    }
}
```

Paired with some kind directory observer, this will allow us to download all files for external locations and then treat them as local.

**Comment by jbmorley** [comment]
'Fixed' in 7bee7b555f1ebed2f6e7c6ef7c7076b965b57a2c.

This change ensures that we automatically trigger a download for any iCloud Drive files (ubiquitous items in Apple API terminology) whenever we detect a directory change. That then triggers a subsequent directory change and the files will magically appear, all without the users intervention (hopefully).

Since we're doing things that might be a little unexpected to the user, we also give users a warning when adding an iCloud Drive location to give them the opportunity to find a smaller sub-directory to add so we don't inadvertently fill their disk.
